docs/ABI.rst

```diff
@@ -1039,7 +1039,7 @@ Types
   type-list ::= list-type '_' list-type*     // list of types
   type-list ::= empty-list
 
-  list-type ::= type identifier? 'z'? 'd'?   // type with optional label, inout convention and variadic specifier
+  list-type ::= type identifier? 'z'? 'h'? 'd'?   // type with optional label, inout convention, shared convention, and variadic specifier
```

Comment:
Basically this is ok. But we have to be careful about filling up the operator namespace.
If we expect that shared is only rarely used, it's not worth "wasting" a whole character 'h'.
Instead we could do something like using 'dv' as a variadic marker (which also does not occur too often) and use 'ds' for the shared convention.